core/src/main/java/nu/yona/server/goals/service/GoalService.java

```diff
@@ -108,19 +114,15 @@ public GoalDTO updateGoal(UUID userID, UUID goalID, GoalDTO newGoalDTO, Optional
 		User userEntity = userService.getUserEntityByID(userID);
 		Goal existingGoal = getGoalEntity(userEntity, goalID);
 
-		if (newGoalDTO.getCreationTime().isPresent())
+		verifyGoalUpdate(existingGoal, newGoalDTO);
+		if (newGoalDTO.getCreationTime().isPresent() && !newGoalDTO.isGoalChanged(existingGoal))
```

Comment:
I see that this is necessary for the test, however, I think that it is good to do a check that the passed creation time is later than the creation time of the existing goal to avoid strange errors at Mobiquity...